examples/vision/bit.py

```diff
@@ -0,0 +1,322 @@
+"""
+Title: Image Classification using BigTransfer(BiT)
+Author: [Sayan Nath](https://twitter.com/sayannath2350)
+Date created: 2021/08/29 
+Last modified: 2021/08/29
+Description: BigTransfer (BiT) State-of-the-art transfer learning for image classification.
+"""
+
+"""
+## Introduction
+
+
+BigTransfer known as BiT is a State-of-the-art transfer learning method for Image Classification Tasks. a set of pre-trained image models that can be transferred to obtain excellent performance on new datasets, even with only a few examples per class. BiT performs well across a surprisingly wide range of data regimes — from 1 example per class to 1M total examples. A detailed analysis of the main components are described [here](https://arxiv.org/pdf/1912.11370.pdf) which leads to high transfer performance.
+"""
+
+"""
+## Setup
+"""
+
+import re
+import numpy as np
+import pandas as pd
+import requests, zipfile, io
+import matplotlib.pyplot as plt
+from imutils import paths
+from pprint import pprint
+from collections import Counter
+from sklearn.preprocessing import LabelEncoder
+
+import tensorflow as tf
+from tensorflow import keras
+import tensorflow_hub as hub
+from tensorflow.keras.utils import to_categorical
+
+SEEDS = 42
+
+np.random.seed(SEEDS)
+tf.random.set_seed(SEEDS)
+
+"""
+## Download the dataset
+
+We will be using Image-Scene-Classification dataset which comprises of 30 classes.
+"""
+
+
+def download_dataset(url):
+    r = requests.get(url)
+    z = zipfile.ZipFile(io.BytesIO(r.content))
+    z.extractall("./")  # Extracting the Data from the zip file
+
+
+download_dataset(
+    "http://data.vision.ee.ethz.ch/ihnatova/camera_scene_detection_train.zip"
+)
+
+"""
+## Data Parsing
+"""
+
+image_paths = list(paths.list_images("training"))
+np.random.shuffle(image_paths)
+image_paths[:5]
+
+"""
+## Counting number of images per class
+"""
+
+labels = []
+for image_path in image_paths:
+    label = image_path.split("/")[1]
+    labels.append(label)
+class_count = Counter(labels)
+pprint(class_count)
+
+"""
+As there is an unequal distribution of classes in the dataset. It is classified as imbalance dataset
+"""
+
+"""
+## Define hyperparameters
+"""
+
+RESIZE_TO = 260
+CROP_TO = 224
+TRAIN_SPLIT = 0.9
+BATCH_SIZE = 128
+AUTO = tf.data.AUTOTUNE
+NUM_CLASSES = 30
+STEPS_PER_EPOCH = 140
+
+"""
+## Splitting the Dataset
+"""
+
+TRAIN_LENGTH = int(len(image_paths) * TRAIN_SPLIT)
+
+train_paths = image_paths[:TRAIN_LENGTH]
+train_labels = labels[:TRAIN_LENGTH]
+validation_paths = image_paths[TRAIN_LENGTH:]
+validation_labels = labels[TRAIN_LENGTH:]
+
+print(len(train_paths), len(validation_paths))
+
+"""
+## Encding the Labels
+"""
+
+label_encoder = LabelEncoder()
```

Comment:
Prefer doing this by hand rather than using sklearn to do it, that will be more explicit

Reply:
Trying to do that.

Reply:
I guess this can add some complicity in the code. As this tutorial is not focused on Label Encoding. What do u think?

Reply:
<img width="1234" alt="Screenshot 2021-09-13 at 1 25 01 AM" src="https://user-images.githubusercontent.com/41967348/133000964-29744f19-cf98-4df3-8c1f-3f464c68d94b.png">

While using `IntegerLookup`